lib/call.js

```diff
@@ -2714,6 +2754,10 @@ class call {
     this._sethangupcause( "us", reason )
 
     if( this.state.established ) {
```

Comment:
Something we might have yet to discuss is early negotiation. In SIP, we can establish an audio channel when either the call is "established" or not "established", and we want to play a tone to the other end to indicate something (this is an "early" state).

I think the test for playing a tone is

If there is a channel AND 
1. If it is established OR
2. If it is early

If you play a tone and then move onto line 2762 (destroy the dialogue), then the call will be destroyed as well - if we want to play a tone - then we have to play the tone and return (this will leave the caller to then hang up after hearing the tone).

On reflection, we should have a flag in the hangup function to indicate this is what we want, as it might have other harmful side effects. i.e.

```js
async hangup( reason, tone = false ) {
```